### ISSUE TITLE
codespell failure

### ISSUE THREAD

**Issue by m-abubakar** [nvimtools/none-ls.nvim#83]
### FAQ

- [X] I have checked the [FAQ](https://github.com/nvimtools/none-ls.nvim#faq) and it didn't resolve my problem.

### Issues

- [X] I have checked [existing issues](https://github.com/nvimtools/none-ls.nvim/issues?q=is%3Aissue) and there are no issues with the same problem.

### Neovim Version

v0.10.0-dev-2363+gb76a01055

### Dev Version?

- [ ] I am using a stable Neovim release version, or if I am using a dev version of Neovim I have confirmed that my issue is reproducible on a stable version.

### Operating System

Both Windows and Linux

### Minimal Config

```Lua
default LunarVim
```

### Steps to Reproduce

codespell fails when words have special characters in them. i.e: Re**-**arrange

```Log
[null-ls] failed to run generator: ...e-ls.nvim/lua/null-ls/builtins/diagnostics/codespell.lua:37: attempt to perform arithmetic on local 'end_col' (a nil value)
```

I have been using a patch for quite some time. So I thought I should share it 😊

### Reproducibility Check

- [X] I confirm that my minimal config is based on the `minimal_init.lua` template and that my issue is reproducible by running `nvim --clean -u minimal_init.lua` and following the steps above.

### Expected Behavior

Special characters should be escaped.

### Actual Behavior

Error when Special characters in words

### Debug Log

```Log
[null-ls] failed to run generator: ...e-ls.nvim/lua/null-ls/builtins/diagnostics/codespell.lua:37: attempt to perform arithmetic on local 'end_col' (a nil value)
```

### Help

Yes

### Requirements

- [X] I have read and followed the instructions above and understand that my issue will be closed if I did not provide the required information.

**Comment by mochaaP** [comment]
I didn't see a patch. Is it included here?

**Comment by mochaaP** [comment]
<https://github.com/nvimtools/none-ls.nvim/blob/b4bd764cd1705086de4bd89f7ccf9d9ed0401259/lua/null-ls/builtins/diagnostics/codespell.lua?plain=1#L27>

It might be an issue with the regex, but since I don't use codespell I have no idea about this.
Also, this builtin has been unmaintained since 2021, so any help is appreciated :)

/cc the original author @Cenobantur

**Comment by m-abubakar** [comment]
Sorry, I got a little sick and had to rest for a while.

I have attached the PR now. Please verify and let me know if you see any improvements can be made. ☺️

**Comment by mochaaP** [comment]
Closed by #84
